packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -709,6 +709,10 @@ Future<void> buildPluginsAsAar(
     assert(pluginDirectory.existsSync());
 
     final String pluginName = pluginParts.first;
+    if (!pluginDirectory.childDirectory('android').existsSync()) {
```

Comment:
nit: should we check for a `android/build.gradle` instead to trigger less FNs?

Reply:
well. detecting a valid Gradle project requires checking additional files. Is that what we want or just Android-platform support?

Reply:
I see your point, I mentioned it as I thought we'd want to be consistent with what we've done here: https://github.com/flutter/flutter/pull/40640